chain/client/src/stateless_validation/chunk_validator/mod.rs

```diff
@@ -208,10 +209,13 @@ pub(crate) fn send_chunk_endorsement_to_block_producers(
 
     // Send the chunk endorsement to the next NUM_NEXT_BLOCK_PRODUCERS_TO_SEND_CHUNK_ENDORSEMENT block producers.
     // It's possible we may reach the end of the epoch, in which case, ignore the error from get_block_producer.
+    // It is possible that the same validator appears multiple times in the upcoming block producers,
+    // thus we collect the block producers in a set.
     let block_height = chunk_header.height_created();
-    let block_producers = (0..NUM_NEXT_BLOCK_PRODUCERS_TO_SEND_CHUNK_ENDORSEMENT)
+    let block_producers: HashSet<AccountId> = (0
+        ..NUM_NEXT_BLOCK_PRODUCERS_TO_SEND_CHUNK_ENDORSEMENT)
         .map_while(|i| epoch_manager.get_block_producer(&epoch_id, block_height + i).ok())
-        .collect_vec();
+        .collect();
```

Comment:
HashSet will randomize the order of block producers. I would feel a bit better if we send it in the order of heights, just to make sure we're not adding any extra latency on the critical path.

Reply:
Itertools has a [unique](https://docs.rs/itertools/0.7.4/itertools/trait.Itertools.html#method.unique) function to do this.

Reply:
better, thanks. done.